src/components/dropdown/dropdown.component.ts

```diff
@@ -46,33 +46,42 @@ export class DropdownComponent implements AfterContentInit, OnDestroy {
   @ContentChild(DropdownMenuDirective)
   dropdownMenu: DropdownMenuDirective;
 
-  private toggleListener: any;
   private documentListener: any;
 
+  constructor(private renderer: Renderer2, private cd: ChangeDetectorRef) {
+  }
+
   ngAfterContentInit(): void {
     this.dropdownToggle.toggle.subscribe(ev => this.onToggleClick(ev));
   }
 
   ngOnDestroy(): void {
-    if(this.toggleListener) this.toggleListener();
     if(this.documentListener) this.documentListener();
   }
 
-  @HostListener('document:click', ['$event'])
   onDocumentClick({ target }): void {
     if(this.open && this.closeOnOutsideClick) {
       const isToggling = this.dropdownToggle.element.contains(target);
       const isMenuClick = !this.closeOnClick && this.dropdownMenu.element.contains(target);
 
       if(!isToggling && !isMenuClick) {
         this.open = false;
+        if (this.documentListener) this.documentListener();
+        this.cd.markForCheck();
       }
     }
   }
 
   onToggleClick(ev): void {
     if(!this.dropdownToggle.disabled) {
       this.open = !this.open;
+
+      if (this.open) {
+        this.documentListener = this.renderer.listen(
```

Comment:
You should look at using RX for listening to the stream of click events.

Reply:
I'll look into it... should be done across components.